src/validation.cpp

```diff
@@ -4194,6 +4194,15 @@ arith_uint256 CalculateClaimedHeadersWork(std::span<const CBlockHeader> headers)
     return total_work;
 }
 
+/** Check the timestamp of the block header is not too far in the future. **/
+static bool CheckHeaderNotFuture(const CBlockHeader& block, BlockValidationState& state)
```

Comment:
In commit cddc0e2a0f2b8f19ef243edc8a85e206c31f9e62:
Can you add some context to the commit message for why this move is being done? Maybe:
"A call to ContextualCheckBlockHeader is added to ConnectBlock in the following commits, but it should not re-check the timestamp of the header there. Move the timestamp check into a separate function to avoid this."

Reply:
Done.